wrangler-core/src/test/java/co/cask/wrangler/lineage/LineageTest.java

```diff
@@ -0,0 +1,473 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.wrangler.lineage;
+
+import co.cask.wrangler.TestingRig;
+import co.cask.wrangler.api.Executor;
+import co.cask.wrangler.api.RecipeParser;
+import co.cask.wrangler.api.lineage.LineageGenerationException;
+import co.cask.wrangler.api.lineage.MutationDefinition;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class LineageTest {
+  private static final boolean PRINT = false;
+
+  private void assertBranchingStepNode(FieldLevelLineage.BranchingTransformStepNode node,
+                                       int stepNumber, boolean back, boolean forward,
+                                       Map<String, Integer> impacting, Map<String,Integer> impacted) {
+    Assert.assertEquals(node.getTransformStepNumber(), stepNumber);
+    Assert.assertEquals(node.continueBackward(), back);
+    Assert.assertEquals(node.continueForward(), forward);
+    Assert.assertEquals(node.getImpactingBranches(), impacting);
+    Assert.assertEquals(node.getImpactedBranches(), impacted);
+  }
+
+  private void print(WranglerFieldLevelLineage lineage, List<String> inputSchema, List<String> outputSchema) {
+    if (PRINT) {
+      System.out.println("\n" + lineage);
+      for (String column : inputSchema) {
+        System.out.println();
+        lineage.prettyPrint(column, true);
+      }
+      for (String column : outputSchema) {
+        System.out.println();
+        lineage.prettyPrint(column, false);
+      }
+    }
+  }
+
+  @Test
+  public void testBasicModifyLineage() throws Exception {
+    String[] directives = new String[] {"lowercase owner"};
+
+    List<String> inputSchema = ImmutableList.of("name", "owner", "species", "sex", "birth", "name2");
+    List<String> outputSchema = ImmutableList.of("name", "owner", "species", "sex", "birth", "name2");
+
+    RecipeParser parser = TestingRig.parse(directives);
+    List<MutationDefinition> parseTree = new ArrayList<>();
+    for (Executor directive : parser.parse()) {
+      parseTree.add(directive.lineage());
+    }
+    WranglerFieldLevelLineage lineage = (new WranglerFieldLevelLineage.Builder(inputSchema, outputSchema)).build(parseTree);
+    Assert.assertEquals(6, lineage.getLineage().size());
+    Assert.assertEquals(1, lineage.getLineage().get("owner").size());
+    assertBranchingStepNode(lineage.getLineage().get("owner").get(0), 0, true, true,
+                            ImmutableMap.<String, Integer>of(), ImmutableMap.<String, Integer>of());
+    print(lineage, inputSchema, outputSchema);
```

Comment:
It is disabled with the static variable at the top, but can be turned on just to get a better visual of the lineage trees.